### PR TITLE
Operation: Implement socket-activatability for HTTP endpoints

### PR DESCRIPTION
**Pull request by cherti**
This commit implement the ability to socket-activate Grafana. This way,
one can use either xinetd's or systemd's socket activation mechanism while
maintaining the current behavior if it is not used.
This has a multitude of advantages, among these:
  * allowing to transparently restart Grafana without breaking
  connections (including transparent upgrades and crash recovery, should it ever
  happen) as the socket is not managed by the Grafana binary
  * allowing to parallelize the start of Grafana and components depending on it,
  as connectivity is already established, allowing for faster boot-times of
  system and/or service complexes
  * allowing on-demand starting of Grafana upon the first connection

### CODE REVIEW

File: docs/sources/installation/rpm.md [hunk 1]
```diff
@@ -164,6 +164,8 @@ Configure the Grafana server to start at boot:
 sudo systemctl enable grafana-server
 ```
 
+Grafana also supports systemd's socket activation mechanism if desired.
```
Comment (oddlittlebird) [inline docs/sources/installation/rpm.md:167]:
```suggestion
Grafana also supports the systemd socket activation mechanism.
```

File: docs/sources/installation/debian.md [hunk 2]
```diff
@@ -127,6 +127,8 @@ Configure the Grafana server to start at boot:
 sudo systemctl enable grafana-server.service
 ```
 
+Grafana also supports systemd's socket activation mechanism if desired.
```
Comment (oddlittlebird) [inline docs/sources/installation/debian.md:130]:
```suggestion
Grafana also supports the systemd socket activation mechanism.
```

**Comment by oddlittlebird** [comment]
Had a couple minor edits for the docs, otherwise that portion of this PR looks fine.

**Comment by cherti** [comment]
Good proposal, rephrased accordingly.

**Comment by oddlittlebird** [comment]
@bergquist , @daniellee , @jessabe , please have someone take a look at this.

**Comment by bergquist** [comment]
Thank you for taking the time to contribute to Grafana but I've decided to close this. 

While the code change might seem small its introduces quite a lot of maintenance burden for something I don't think is commonly used. I've opened an issue for it and if it gets enough attention we will reconsider this. 

https://github.com/grafana/grafana/issues/25885

**Pull request closed without merging**